### PR TITLE
Upgrade tool management

### PR DESCRIPTION
**Pull request by wagoodman**
Upgrade practices similar to that of syft and grype, specifically:
- use binny to manage local tooling
- use task over make for coordinating tasks

This helps setup getting a release pipeline into stereoscope #221 , since it will be quite similar to how syft and grype release workflow-wise, getting the practices and tooling up to date first makes the transition easier.

### CODE REVIEW

File: .github/actions/bootstrap/action.yaml [hunk 1]
```diff
@@ -1,80 +1,60 @@
...
+
 description: "Bootstrap all tools and dependencies"
 inputs:
   go-version:
     description: "Go version to install"
     required: true
-    default: "1.19.x"
-  use-go-cache:
-    description: "Restore go cache"
+    default: "1.21.x"
```
Comment (spiffcs) [inline .github/actions/bootstrap/action.yaml:8]: Is 1.22 available yet?

File: Makefile [hunk 2]
```diff
@@ -1,208 +1,46 @@
-TEMP_DIR = ./.tmp
+OWNER = anchore
+PROJECT = syft
```
Comment (spiffcs) [inline Makefile:2]: I think this needs to be `stereoscope`

File: Taskfile.yaml [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+
+  # note: the snapshot dir must be a relative path starting with ./
+  SNAPSHOT_DIR: ./snapshot
+  SNAPSHOT_CMD: "{{ .TOOL_DIR }}/goreleaser release --config {{ .TMP_DIR }}/goreleaser.yaml --clean --snapshot --skip=publish --skip=sign"
+  RELEASE_CMD:  "{{ .TOOL_DIR }}/goreleaser release --clean --release-notes {{ .CHANGELOG }}"
+  VERSION:
+    sh: git describe --dirty --always --tags
+
+  # used for install and acceptance testing
+  COMPARE_DIR: ./test/compare
```
Comment (spiffcs) [inline Taskfile.yaml:23]: Are these used by stereoscope at all?

**Review by spiffcs — COMMENTED** [review]
small comments added with questions about use of variables and potential project name conflict

File: Makefile (hunk 2 above)
Comment (wagoodman) [reply Makefile:2]: nice find

File: Taskfile.yaml (hunk 3 above)
Comment (wagoodman) [reply Taskfile.yaml:23]: no,I'll remove these... I'll do one more pass for syft-specific things that got copied in

**Review by spiffcs — APPROVED** [review]
👍 with comments addressed

**Pull request merged**